public/src/components/ToastManager.jsx

```diff
@@ -0,0 +1,198 @@
+import { useState, useRef } from 'react';
+import * as React from 'react';
+import { PropTypes } from 'prop-types';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faCheckCircle, faExclamationTriangle, faTimes,
+} from '@fortawesome/free-solid-svg-icons';
+
+const getCSSForFlavor = (flavor) => {
+  const flavors = {
+    success: `
+      background-color: #386C0B;
+    `,
+    error: `
+      background-color: #EF6F6C;
+    `,
+  };
+  return flavors[flavor];
+};
+
+export const ToastContext = React.createContext();
+const ToastRack = styled.div`
+  z-index: 1;
+  position: fixed;
+  right: 0;
+  width: 250px;
+  bottom: 0;
+  padding: 1em;
+`;
+
+const StyledToast = styled.div`
+  @keyframes styledtoast {
+    0% {
+      transform: translateY(20px)
+    }
+    100% {
+      transform: translateY(0px)
+    }
+  }
+  @keyframes cancelled {
+  0% {
+    transform: translateX(0px);
+  }
+  90% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+  @keyframes immediatecancel {
+  0% {
+    transform: translateX(0px);
+  }
+  100% {
+    transform: translateX(300px);
+  }
+}
+
+  display: flex;
+  border: 1px solid #54457F;
+  width: 250px;
+  cursor: pointer;
+  animation: styledtoast .3s ease-out, cancelled 4.5s ease-out .3s;
+  animation-fill-mode: forwards;  
+  border-radius: .25em;
+  background-color: white;
+  box-shadow: 0px 4px 3px 3px rgba(0,0,0, .1);
+  & + & {
+    margin-top: 1em;
+  }
+
+  & .toast__alert {
+    margin-left: -1px;
+    border-radius: .25em 0 0 .25em;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 25px;
+    & path {
+      fill: white;
+    }
+    padding: .25em .5em;
+    ${(props) => getCSSForFlavor(props.flavor)}
+  }
+  & .toast__content {
+    padding: .5em 1em;
+    flex: 1;
+  }
+  &.toast--cancelled {
+    animation: immediatecancel .35s ease-out;
+  }
+  & .toast__cancel {
+    display: flex;
+    padding: .25em .5em;
+    & path {
+      fill: #54457F;
+    }
+    align-items: center;
+  }
+`;
+
+export const ToastAlert = ({ children, flavor, id }) => {
+  const { manuallyCancel, flavors } = React.useContext(ToastContext);
+  const internalFlavors = {
+    [flavors.success]: <FontAwesomeIcon icon={faCheckCircle} />,
+    [flavor.error]: <FontAwesomeIcon icon={faExclamationTriangle} />,
+  };
+  const alertIcon = internalFlavors[flavor];
```

Comment:
nit: no need to define this here, do it in the jsx

Reply:
ergo 

```jsx

<StyledToast
      flavor={flavor}
      ariaLive="polite"
      className={cancel ? 'toast--cancelled' : ''}
      onClick={handleClick}
    >
      <div className="toast__alert">
        {internalFlavors[flavor];}
      </div>
      <div className="toast__content">
        {children}
      </div>
      <div className="toast__cancel">
        <FontAwesomeIcon icon={faTimes} />
      </div>
    </StyledToast>
```

?

Reply:
sans the semi-colon, yes.

Reply:
Done.